lib/recurly/pricing/checkout/index.js

```diff
@@ -210,30 +246,69 @@ export default class CheckoutPricing extends Pricing {
         }
       }
 
-      let adjustment = {};
-
-      if (existingAdjustment) {
-        // Only pick present attributes
-        if (typeof amount !== 'undefined') adjustment.amount = amount;
-        if (typeof currency !== 'undefined') adjustment.currency = currency;
-        if (typeof quantity !== 'undefined') adjustment.quantity = parseInt(quantity, 10);
-        if (typeof taxExempt !== 'undefined') adjustment.taxExempt = taxExempt;
-        if (typeof taxCode !== 'undefined') adjustment.taxCode = taxCode;
-
-        Object.assign(existingAdjustment, adjustment);
-      } else {
-        // Adjustment defaults
-        quantity = parseInt(quantity, 10);
-        if (isNaN(quantity)) quantity = 1;
-        currency = currency || this.items.currency;
-        taxExempt = !!taxExempt;
+      if (existingAdjustment && existingAdjustment.itemCode && !itemCode) {
+        itemCode = existingAdjustment.itemCode;
+      }
 
-        adjustment = { amount, quantity, id, currency, taxExempt, taxCode };
+      new Promise((resolve, reject) => {
+        if (!itemCode) return resolve();
+        return recurly.item({ itemCode }).then(item => {
+          const expectedCurrency = currency
+                                || (existingAdjustment && existingAdjustment.currency)
+                                || this.items.currency;
+          if (typeof amount === 'undefined') {
+            const itemPrice = find(item.currencies, c => c.currency_code === expectedCurrency);
+            if (itemPrice) {
+              amount = itemPrice.unit_amount;
+            } else {
+              return reject(errors('invalid-item-currency', { itemCode, currency: expectedCurrency }));
+            }
+          }
```

Comment:
`amount` is the only value that may be overridden when an `itemCode` is specified.